arrow-meta/src/main/kotlin/arrow/meta/plugins/proofs/phases/ProofsExt.kt

```diff
@@ -0,0 +1,63 @@
+package arrow.meta.plugins.proofs.phases
+
+import org.jetbrains.kotlin.descriptors.CallableMemberDescriptor
+import org.jetbrains.kotlin.descriptors.ClassConstructorDescriptor
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ClassKind
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.descriptors.annotations.AnnotationDescriptor
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.resolve.calls.util.FakeCallableDescriptorForObject
+import org.jetbrains.kotlin.resolve.descriptorUtil.isExtension
+
+fun DeclarationDescriptor.contextualAnnotations(): Set<FqName> =
+  annotations.mapNotNull {
+    if (it.isGivenContextProof()) it.fqName
+    else null
+  }.toSet()
+
+fun DeclarationDescriptor.asProof(): Sequence<Proof> =
+  when (this) {
+    is PropertyDescriptor -> asProof()
+    is ClassConstructorDescriptor -> containingDeclaration.asProof()
+    is FunctionDescriptor -> asProof()
+    is ClassDescriptor -> asProof()
+    is FakeCallableDescriptorForObject -> classDescriptor.asProof()
+    else -> TODO("asProof: Unsupported proof declaration type: $this")
```

Comment:
Can we provide more information for the user here?
Ideally we provide the code location, line-number with file.

Same for other in this file, but can be moved to ticket/seperate PR.

Reply:
I think at some point, this got refactored to become a `Sequence<Proof>`, and the `else` case should have instead returned an empty sequence since the cases above are all the kind of declarations we currently support as proofs. I'll change it to return an emptySequence